### DISCUSSION TITLE
Why do struct fields such as `status` and `spec` have `omitempty` tag?

### DISCUSSION THREAD

**Discussion by therealak12** [kubernetes-sigs/kubebuilder#3087] [Support] [answered]
The generated code contains an `omitempty` tag for top-level struct fields but AFAIK it doesn't have any effect for struct fields.

- Why do we add this tag when it has no effect?
- What if we change the type of status field into pointer?

`` Status ApplicationStatus `json:"status,omitempty" ``

**Comment by camilamacedo86** [comment, marked as answer]
Hi @therealak12,

That means that the spec value is not mandatory and can be empty.
Please, see: https://book.kubebuilder.io/reference/markers.html?highlight=omitempty#difference-between--optional-and--kubebuildervalidationoptional

**Reply by therealak12** [reply]
Hi @camilamacedo86 

Thanks!

**Comment by camilamacedo86** [comment]
Closing since it is answered.
